cellstore/reader.go

```diff
@@ -176,6 +178,84 @@ func (r *Reader) readBlock(pos int) ([]byte, error) {
 	}
 }
 
+type Near struct {
+	CellID   s2.CellID
+	Value    []byte
+	Distance int32
+}
+
+type Nearby []*Near
+type ByDistance []*Near
+
+func (n ByDistance) Len() int           { return len(n) }
+func (n ByDistance) Swap(i, j int)      { n[i], n[j] = n[j], n[i] }
+func (n ByDistance) Less(i, j int) bool { return n[i].Distance < n[j].Distance }
+
+// FindNearby
+func (r *Reader) FindNearby(loc s2.CellID, limit int) (Nearby, error) {
+	var dst Nearby
```

Comment:
the process is fine, but I am not sure it's what we want, I think we want to return our own `Iterator` object with the relevant blocks combined instead. to do that you need to concat the buffers of the individual iterators